SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -1359,50 +1220,23 @@ void CSolver::CompletePeriodicComms(CGeometry *geometry,
               break;
 
             case PERIODIC_SOL_GG:
-
-              /*--- For G-G, we accumulate partial gradients then compute
-               the final value using the entire volume of the periodic cell. ---*/
-
-              for (iVar = 0; iVar < nVar; iVar++)
-                for (iDim = 0; iDim < nDim; iDim++)
-                  base_nodes->AddGradient(iPoint, iVar, iDim,
-                                          bufDRecv[buf_offset+iVar*nDim+iDim]);
-
-              break;
-
+            case PERIODIC_SOL_GG_R:
             case PERIODIC_PRIM_GG:
+            case PERIODIC_PRIM_GG_R:
 
               /*--- For G-G, we accumulate partial gradients then compute
                the final value using the entire volume of the periodic cell. ---*/
 
-              for (iVar = 0; iVar < nPrimVarGrad; iVar++)
+              for (iVar = 0; iVar < ICOUNT; iVar++)
                 for (iDim = 0; iDim < nDim; iDim++)
-                  base_nodes->AddGradient_Primitive(iPoint, iVar, iDim,
-                                                    bufDRecv[buf_offset+iVar*nDim+iDim]);
-              break;
-
-            case PERIODIC_SOL_LS: case PERIODIC_SOL_ULS:
-
-              /*--- For L-S, we build the upper triangular matrix and the
-               r.h.s. vector by accumulating from all periodic partial
-               control volumes. ---*/
-
-              for (iDim = 0; iDim < nDim; iDim++) {
-                for (jDim = 0; jDim < nDim; jDim++) {
-                  base_nodes->AddRmatrix(iPoint, iDim,jDim,bufDRecv[buf_offset]);
-                  buf_offset++;
-                }
-              }
-              for (iVar = 0; iVar < nVar; iVar++) {
-                for (iDim = 0; iDim < nDim; iDim++) {
-                  base_nodes->AddGradient(iPoint, iVar, iDim, bufDRecv[buf_offset]);
-                  buf_offset++;
-                }
-              }
+                  (*gradient)(iPoint, iVar, iDim) += bufDRecv[buf_offset+iVar*nDim+iDim];
 
               break;
 
+            case PERIODIC_SOL_LS: case PERIODIC_SOL_ULS:
+            case PERIODIC_SOL_LS_R: case PERIODIC_SOL_ULS_R:
             case PERIODIC_PRIM_LS: case PERIODIC_PRIM_ULS:
+            case PERIODIC_PRIM_LS_R: case PERIODIC_PRIM_ULS_R:
```

Comment:
We have more comm types but less comm code, because a gradient is a gradient and a limiter is a limiter.
If you review these changes be on the look out for uses of nVar and nPrimVarGrad instead of something that looks more generic like COUNT_PER_POINT, ICOUNT, or nVarGrad (I have introduced bugs like that in the past with this type of refactoring...)

Reply:
And you tell me that now that I am close to the end of the diff... ok I scroll through it again